### PR TITLE
extends user cluster provider component

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: simply extends user cluster provider component so that  additional options to the rest config can be added in a generic way.
 
**Special notes for your reviewer**: see #2680 for more info

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
> pkg/cluster/client/client.go:45:6: type name will be used as client.ClientConfigOption by other packages, and that stutters; consider calling this ConfigOption (golint)
type ClientConfigOption func(*restclient.Config) *restclient.Config
     ^

Dear `golint` don't you dare to educate me about naming conventions in our project.

File: api/pkg/cluster/client/client.go [hunk 1]
```diff
@@ -41,8 +41,11 @@ func (p *Provider) GetAdminKubeconfig(c *kubermaticv1.Cluster) ([]byte, error) {
 	return d, nil
 }
 
+// ClientConfigOptions defines a function that applies additional configuration to restclient.Config in a generic way.
```
Comment (zreigz) [inline api/pkg/cluster/client/client.go]: change to `ConfigOption`

File: api/pkg/provider/types.go [hunk 2]
```diff
@@ -107,15 +107,15 @@ type ClusterProvider interface {
...
 	//
 	// Note that the client you will get has admin privileges
-	GetMachineClientForCustomerCluster(cluster *kubermaticv1.Cluster) (clusterv1alpha1clientset.Interface, error)
+	GetAdminMachineClientForCustomerCluster(cluster *kubermaticv1.Cluster) (clusterv1alpha1clientset.Interface, error)
 
 	// GetClientForCustomerCluster returns a client to interact with the given cluster
 	//
 	// Note that the client you will get has admin privileges
-	GetKubernetesClientForCustomerCluster(cluster *kubermaticv1.Cluster) (kubernetes.Interface, error)
+	GetAdminKubernetesClientForCustomerCluster(cluster *kubermaticv1.Cluster) (kubernetes.Interface, error)
```
Comment (zreigz) [inline api/pkg/provider/types.go:118]: change doc name `GetClientForCustomerCluster` -> `GetAdminKubernetesClientForCustomerCluster`

**Review by zreigz — CHANGES_REQUESTED** [review]
rest looks good

**Review by zreigz — APPROVED** [review]
/lgtm

**Comment by p0lyn0mial** [comment]
@mrIncompetent PTAL

**Pull request merged**